### PR TITLE
Card notification icons

### PR DESCRIPTION
**Pull request by Showcas**
adds notification icons for the cards in the top right

the price is only shown on the preview cards for the search

are you happy with the icon choices?

### CODE REVIEW

**Comment by fynsta** [comment]
I don't love the way it looks (and the way the information is just redundant), but I cannot come up with a better way right now... The icons are fine, but they just feel a little out of place.

File: lib/util/trip/ride_card.dart [hunk 1]
```diff
@@ -92,7 +92,42 @@ class _RideCardState extends TripCardState<Ride, RideCard> {
...
+          );
+        case RideStatus.approved:
+          return Icon(
+            Icons.done_all,
+            color: pickStatusColor(),
+            key: const Key('approved'),
+          );
+        case RideStatus.rejected:
+          return Icon(
+            Icons.close,
```
Comment (fynsta) [inline lib/util/trip/ride_card.dart:117]: Icons.block makes more sense here, and use something else for cancellation?

File: lib/util/trip/drive_card.dart [hunk 2]
```diff
@@ -72,6 +72,33 @@ class _DriveCardState extends TripCardState<Drive, DriveCard> {
...
+  Widget buildTopRight() {
+    if (!_fullyLoaded) {
+      return const Center(
+        child: SizedBox(),
+      );
+    } else if (_drive.cancelled) {
+      return Icon(
+        Icons.block,
+        color: pickStatusColor(),
+        key: const Key('cancelled'),
```
Comment (fynsta) [inline lib/util/trip/drive_card.dart:85]: the key should contain the fact that this is a statusindicator

File: test/widgets/cards_test.dart [hunk 3]
```diff
@@ -259,9 +259,6 @@ void main() {
 
       //Rating is shown
       expect(find.byType(CustomRatingBarIndicator), findsOneWidget);
-
-      //Price is shown
-      expect(find.byKey(const Key('price')), findsOneWidget);
```
Comment (fynsta) [inline test/widgets/cards_test.dart:264]: this seems unrelated?

File: test/widgets/cards_test.dart [hunk 4]
```diff
@@ -420,26 +431,30 @@ void main() {
...
         await loadDriveCard(tester, drive);
 
+        expect(find.byKey(const Key('cancelled')), findsOneWidget);
+
         final BuildContext context = tester.element(find.byType(Container).first);
         expect(tester.widget<Card>(find.byType(Card)).color, Theme.of(context).colorScheme.error);
       });
 
-      testWidgets('waring color for drive with ride requests', (WidgetTester tester) async {
+      testWidgets('waring status for drive with ride requests', (WidgetTester tester) async {
```
Comment (fynsta) [inline test/widgets/cards_test.dart:448]: ofc not related to your PR, but I just noticed: waring -> warning

File: test/widgets/cards_test.dart (hunk 3 above)
Comment (Showcas) [reply test/widgets/cards_test.dart:264]: the price was before there where the icon was and the price is only shown for preview status

Comment (fynsta) [reply test/widgets/cards_test.dart:264]: ah yes, thanks!

File: lib/util/trip/ride_card.dart (hunk 1 above)
Comment (Showcas) [reply lib/util/trip/ride_card.dart:117]: I think its easiest for all cancelled/rejected rides to have the same icon

File: lib/util/trip/drive_card.dart [hunk 5]
```diff
@@ -72,6 +72,33 @@ class _DriveCardState extends TripCardState<Drive, DriveCard> {
     return SeatIndicator(trip);
   }
 
+  @override
+  Widget buildTopRight() {
+    if (!_fullyLoaded) {
+      return const Center(
```
Comment (fynsta) [inline lib/util/trip/drive_card.dart:78]: what does the Center do?

File: lib/util/trip/drive_card.dart [hunk 6]
```diff
@@ -72,6 +72,33 @@ class _DriveCardState extends TripCardState<Drive, DriveCard> {
...
+        child: SizedBox(),
+      );
+    } else if (_drive.cancelled) {
+      return Icon(
+        Icons.block,
+        color: pickStatusColor(),
```
Comment (fynsta) [inline lib/util/trip/drive_card.dart:84]: let's load the color into a variable, we need it in all three cases

File: lib/util/trip/drive_card.dart (hunk 5 above)
Comment (Showcas) [reply lib/util/trip/drive_card.dart:78]: it can be taken out, mb

**Review by MarkFalke — APPROVED** [review]

**Review by istzen — APPROVED** [review]
i aggre with @fynsta, but apart from that everything is fine

File: lib/util/trip/drive_card.dart [hunk 7]
```diff
@@ -72,6 +72,33 @@ class _DriveCardState extends TripCardState<Drive, DriveCard> {
...
+        key: const Key('cancelledIcon'),
+      );
+    } else if (_drive.rides!.any((Ride ride) => ride.status == RideStatus.pending)) {
+      return Icon(
+        Icons.access_time_outlined,
+        color: statusColor,
+        key: const Key('pendingIcon'),
+      );
+    } else {
+      return Icon(
```
Comment (Eddie-42) [inline lib/util/trip/drive_card.dart:94]: IMO, you don't actually need an icon in this case. We don't want to give the impression that the drive is somehow "finished".

**Review by Eddie-42 — COMMENTED** [review]
Sorry for the late review. Otherwise looking good!

File: lib/util/trip/ride_card.dart [hunk 8]
```diff
@@ -92,7 +93,37 @@ class _RideCardState extends TripCardState<Ride, RideCard> {
 
   @override
   Widget buildTopRight() {
-    return Text(key: const Key('price'), ' ${_ride.price}€');
+    if (!_fullyLoaded) {
```
Comment (Eddie-42) [inline lib/util/trip/ride_card.dart:96]: This also applies here, and why did we decide to make past rides green depending on whether they are approved? I think they should be grey. If I remember correctly, you can then still tell if a ride is rejected/cancelled because the whole card is grey.

Comment (Showcas) [reply lib/util/trip/ride_card.dart:96]: past drives that took place are green, past drives taht did not are grey (cancelled)

Comment (Showcas) [reply lib/util/trip/ride_card.dart:96]: To me it is just to make it more understandable, but yea... you think all grey and greyout the ones that did not take place?

File: lib/util/trip/drive_card.dart [hunk 9]
```diff
@@ -72,6 +72,31 @@ class _DriveCardState extends TripCardState<Drive, DriveCard> {
...
+        child: SizedBox(),
+      );
+    } else if (_drive.cancelled) {
+      return Icon(
+        Icons.block,
+        color: statusColor,
+        key: const Key('cancelledIcon'),
+      );
+    } else if (_drive.rides!.any((Ride ride) => ride.status == RideStatus.pending)) {
+      return Icon(
```
[the remark this answers is not in the document]
Comment (Showcas) [reply lib/util/trip/drive_card.dart:88]: But shouldn't the notification icons be an additional idicator. like yes I understnd that it is "trivial". But at least put an icon for drive/rides that took place and for the ones that didn't

File: lib/util/trip/ride_card.dart (hunk 8 above)
Comment (fynsta) [reply lib/util/trip/ride_card.dart:96]: Yes I think so too. Makes it more consistent with Drive

File: lib/util/trip/drive_card.dart [hunk 10]
```diff
@@ -72,6 +72,31 @@ class _DriveCardState extends TripCardState<Drive, DriveCard> {
...
+      );
+    } else {
+      return const Center(
+        child: SizedBox(),
+      );
+    }
+  }
+
   @override
   Color pickStatusColor() {
```
Comment (fynsta) [inline lib/util/trip/drive_card.dart:101]: We talked about this but please make this a getter function

File: lib/util/trip/drive_card.dart (hunk 9 above)
[the remark this answers is not in the document]
Comment (fynsta) [reply lib/util/trip/drive_card.dart:88]: What do you care if there were pending rides for past drives? They are irrelevant

File: lib/util/trip/ride_card.dart (hunk 8 above)
Comment (fynsta) [reply lib/util/trip/ride_card.dart:96]: I would say they are basically rejected from a visual standpoint

**Review by Eddie-42 — APPROVED** [review]
